src/radio/radio-controller.ts

```diff
@@ -0,0 +1,166 @@
+import * as dgram from 'dgram';
+import {UdpResponse} from './dataflow';
+
+/**
+ * A default radio timeout, in milliseconds.
+ */
+const RADIO_TIMEOUT = 2500;
+
+/**
+ * A class to contain all parameters required to perform
+ * a UDP scan.
+ */
+export class UDPScanConfig {
+  broadcastAddress: string;
+  broadcastPort: number;
+  listenPort: number;
+  discoveryPacket: string;
+  /**
+   *
+   * @param broadcastAddress  The destination UDP broadcast address.
+   * @param broadcastPort  The destination UDP broadcast port.
+   * @param listenport  The listen port for the UDP response.
+   * @param discoveryPacket  The payload to send in the UDP broadcast.
+   * @returns  A new `UDPScanConfig` instance.
+   */
+  constructor(
+    broadcastAddress: string,
+    broadcastPort: number,
+    listenport: number,
+    discoveryPacket: string
+  ) {
+    this.broadcastAddress = broadcastAddress;
+    this.broadcastPort = broadcastPort;
+    this.listenPort = listenport;
+    this.discoveryPacket = discoveryPacket;
+  }
+}
+
+/**
+ * A class to contain the information from a UDP scan.
+ */
+export interface UDPScanResults {
+  buffer: Buffer;
+  address: string;
+}
+
+/**
+ * A class to contain all Node radio functionality.
+ */
+export class RadioController {
+  /**
+   * A helper function to create timeout promises.
+   * @returns  A new Promise that resolves after RADIO_TIMEOUS milliseconds.
+   */
+  private createTimeoutPromise(): Promise<void> {
+    return new Promise(resolve => {
+      setTimeout(resolve, RADIO_TIMEOUT);
```

Comment:
Would it make more sense for this to reject rather than resolve?

Reply:
I think it makes sense for the timeout promise to resolve, since that's its purpose.  The `Promise.race()` call rejects after this timeout resolves, which I think is the appropriate place to start the `reject` since up until then it's expected/normal behavior